plasmapy/atomic/particle_input.py

```diff
@@ -127,9 +127,12 @@ def particle_input(wrapped_function: Callable = None,
     Raises
     ------
     `TypeError`
-        If the annotated argument is not a `str`, `int`, or
-        `~plasmapy.atomic.Particle`; or if `Z` or `mass_numb` is
-        not an `int`.
+        If the annotated argument is not a `str`, `int`, `tuple`, `list`
+        or `~plasmapy.atomic.Particle`; or if `Z` or `mass_numb` is
+        not an `int`; or if number of elements in a tuple do not match
+        the expected parameters; or if invalid list annotations in
```

Comment:
Invalid list annotations sounds pretty unclear. However, I'm wondering whether we might be getting too verbose here... I think we can cut some of the description, if someone gets an exception they'll be looking at the code anyway, right?

Reply:
I thought so too. If somebody gets an exception, they are told about the reason they are getting the exception. Maybe we should just have these
> If the annotated argument is not a `str`, `int`, `tuple`, `list` or `~plasmapy.atomic.Particle`; or if `Z` or `mass_numb` is not an `int`.

Not sure though.

Reply:
That's beautiful. Put that in! :)

Reply:
Okie!